app/src/main/java/org/literacyapp/service/FaceRecognitionTrainingJobService.java

```diff
@@ -4,28 +4,27 @@
 import android.app.job.JobService;
 import android.util.Log;
 
-import org.literacyapp.authentication.TrainingHelper;
+import org.literacyapp.authentication.TrainingThread;
 
 /**
  * Created by sladomic on 26.11.16.
  */
 
 public class FaceRecognitionTrainingJobService extends JobService {
-    public static boolean isRunning = false;
+    private TrainingThread trainingThread;
 
     @Override
     public boolean onStartJob(JobParameters jobParameters) {
         Log.i(getClass().getName(), "onStartJob");
-        isRunning = true;
-        TrainingHelper trainingHelper = new TrainingHelper(getApplicationContext());
-        trainingHelper.extractFeatures();
-        trainingHelper.trainClassifier();
+        trainingThread = new TrainingThread(getApplicationContext());
+        trainingThread.start();
         return false;
     }
 
     @Override
     public boolean onStopJob(JobParameters jobParameters) {
         Log.i(getClass().getName(), "onStopJob");
+        trainingThread.interrupt();
```

Comment:
extractFeatures would simply pick up where it left because of the query "....where(StudentImageDao.Properties.StudentImageFeatureId.eq(0)..." and trainClassifier too "....where(StudentImageCollectionEventDao.Properties.MeanFeatureVector.isNull()..."